.github/workflows/pull-request-ci.yaml

```diff
@@ -0,0 +1,307 @@
+    - run: make yamllint
+    - run: make helmlint
+
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    needs: lint
+    env:
+      PINNED_TOOLS: true
+      TOOLS_DIR: ${{ github.workspace }}/tools
+    defaults:
+      run:
+        working-directory: kubecf
+    steps:
+    - name: Checkout KubeCF
+      uses: actions/checkout@v2
+      with:
+        path: kubecf
+        fetch-depth: 0
+        submodules: recursive
+    - uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: cache.tools
+      uses: actions/cache@v2
+      with:
+        path: ${{ github.workspace }}/tools
+        key: ${{ runner.os }}-tools
+    - run: make tools-install
+    - run: make kubecf-bundle
+      env:
+        TARGET_FILE: ${{ github.workspace }}/kubecf-bundle.tgz
+    - run: |
+        mkdir bundle
+        tar xf kubecf-bundle.tgz -C bundle
+      working-directory: ${{ github.workspace }}
+    - name: Upload bundle
+      uses: actions/upload-artifact@v2
+      with:
+        name: kubecf-bundle.tgz
+        path: ${{ github.workspace }}/bundle
+
+  tests:
+    name: tests
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental }}
+    needs: build
+    env:
+      # For KubeCF
+      PINNED_TOOLS: true
+      TOOLS_DIR: ${{ github.workspace }}/tools
+      FEATURE_AUTOSCALER: "true"
+      # For catapult
+      # Note that credentials-related parts are in the individual steps
+      BACKEND: gke
+      DOWNLOAD_CATAPULT_DEPS: "false"
+
+    defaults:
+      run:
+        working-directory: kubecf
+
+    strategy:
+      matrix:
+        include:
+        - backend: diego
+          experimental: false
+        - backend: eirini
+          experimental: true
+
+    steps:
+    - name: Checkout KubeCF
+      uses: actions/checkout@v2
+      with:
+        path: kubecf
+        submodules: recursive
+    - name: Checkout catapult
+      uses: actions/checkout@v2
+      with:
+        repository: SUSE/catapult
+        path: catapult
+    - uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: cache.tools
+      uses: actions/cache@v2
+      with:
+        path: ${{ github.workspace }}/tools
+        key: ${{ runner.os }}-tools
+    - run: make tools-install
+
+    - id: download-bundle
+      name: Download chart bundle
+      uses: actions/download-artifact@v2
+      with:
+        name: kubecf-bundle.tgz
+        # The artifact upload/download creates an extra directory
+        path: ${{ github.workspace }}/kubecf-bundle
+
+    - name: Download gomplate
+      run: |
+        if test -x "${TOOLS_DIR}/gomplate" ; then
+          echo "gomplate already exists; skipping."
+          exit 0
+        fi
+        mkdir -p "${TOOLS_DIR}"
+        wget -O "${TOOLS_DIR}/gomplate" \
+          https://github.com/hairyhenderson/gomplate/releases/latest/download/gomplate_linux-amd64
+        chmod a+x "${TOOLS_DIR}/gomplate"
+
+    - name: Start SSH agent
```

Comment:
Why do we need this? If for debugging, we should use the tmate action [1].

[1] https://github.com/marketplace/actions/debugging-with-tmate

Reply:
This was for debugging and we can drop it.

The tmate action isn't great because
- It's by some random guy; I've tried to limit the actions here to be first-party Microsoft ones (including GitHub).
- it hangs the CI until you connect and create a file.  That's not useful if we just want to be able to get in only if the run fails or looks suspicious.

Reply:
commented. It is a requirement for catapult.

Reply:
@prabalsharma what does catapult use it for?

Reply:
Sorry, I cannot remember now, I will ask around.

Reply:
https://github.com/SUSE/catapult/wiki/AKS#preparation documents the requirement (remember that I started with AKS).